Sources/Datasets/Dataset.swift

```diff
@@ -1,8 +1,24 @@
 protocol S5TFDataset {
     associatedtype _Iterator // swiftlint:disable:this type_name
     func batched(_ batchSize: Int) -> _Iterator
+    var info: S5TFDatasetInfo { get }
 }
 
 protocol S5TFDatasetIterator: Sequence, IteratorProtocol where _Dataset: S5TFDataset {
     associatedtype _Dataset // swiftlint:disable:this type_name
 }
+
+public struct S5TFDatasetInfo {
+	var name = ""
+	var version = ""
+	var description = ""
+	var homepage = ""
+	var citation = ""
+	public init(name: String = "", version: String = "", description: String = "", homepage: String = "", citation: String = "") {
+		self.name = name
+		self.version = version
+		self.description = description
+		self.homepage = homepage
+		self.citation = citation
+	}
+}
```

Comment:
I like how this is a struct. Consider changing it like this though:

```swift
var name: String
var version: String
var description: String
var homepage: String
var citation: String?
```

We are not sure whether or not a citation is available. The other items are required by the contributor of course. You can omit the initializer with this approach because Swift will generate it for you.